xrally_kubernetes/service.py

```diff
@@ -78,3 +154,75 @@ def list_namespaces(self):
                  "uid": r.metadata.uid,
                  "labels": r.metadata.labels}
                 for r in self.v1_client.list_namespace().items]
+
+    @atomic.action_timer("kube.create_namespace")
+    def create_namespace(self, name):
+        """Create namespace.
+
+        :param name: namespace name
+        """
+        manifest = {
+            "apiVersion": "v1",
+            "kind": "Namespace",
+            "metadata": {
+                "name": name,
+                "labels": {
+                    "role": "test-%s" % name
+                }
+            }
+        }
+        self.v1_client.create_namespace(body=manifest)
+        LOG.info("Namespace %s create started" % name)
+
+    @atomic.action_timer("create_namespace_and_wait_active_status")
+    def create_namespace_and_wait_active(self, name, sleep_time=5,
+                                         retries_total=30):
+        """Create namespace and wait until status phase won't be Active.
+
+        :param name: namespace name
+        :param sleep_time: poll interval between each two retries
+        :param retries_total: number of total retries of reading status
+        """
+        self.create_namespace(name)
+
+        LOG.debug("Wait until namespace %s status won't be active" % name)
+        wait_for_status(name,
+                        status="Active",
+                        read_method=self.v1_client.read_namespace,
```

Comment:
First of all,  let's move a call for `self.v1_client.read_namespace` to the separate method with a proper action_timer decorator.

The second issue I see here, it is the final representation of atomic actions. There will be action "create_namespace_and_wait_active_status" which takes X and has one child "create_namespace" which takes Y. By calculations, I can recognize how much "waiting" took, but it is not so obvious.

As I proposed in the first sentence if we wrap read_namespace action, we will already have the more detailed view (with information about a number of read requests). Another thing, let' wrap execution of wait_for_status with atomic action as well, so it the code will look something like:

```python

if wait_for_status: # defaults to True
    with atomic.ActionTimer(self, "kubernetes.wait_for_nc_become_active"): # probably, the better name can be used
        wait_for_status(name,
                        status="Active",
                        read_method=self.get_namespace)
```
In this can we will have something like:

```
kubernetes.create_namespace == 5s
-> kubernetes.wait_for_nc_become_active == 3s
---> kubernetes.get_namespace (x3) == 2s
```